bindinfo/cache.go

```diff
@@ -56,6 +56,14 @@ type Binding struct {
 	id string
 }
 
+func (b *Binding) isSame(rb *Binding) bool {
+	if b.id != "" && rb.id != "" {
+		return b.id == rb.id
+	}
+	// Sometimes we cannot construct `id` because of the changed schema, so we need to compare by bind sql.
+	return b.BindSQL == rb.BindSQL
```

Comment:
So can we directly compare the SQL instead of the id?

Reply:
Users may write hints in different ways but acutally there are the same, so `id` is a way to handle the duplicates. And we we evolve bindings, it is important to not evolve existing bindings.

Reply:
I got you.